gwpy/timeseries/timeseries.py

```diff
@@ -1080,11 +1090,21 @@ def filter(self, *filt, **kwargs):
         # parse filter
         form, filt = filter_design.parse_filter(filt)
 
+        unit = kwargs.pop('unit', False)
```

Comment:
`None` might be a more readable default for this case.

```suggestion
        unit = kwargs.pop('unit', None)
```